docs/sources/operators-guide/deploying-grafana-mimir/jsonnet/memberlist.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "Migrating from Consul to Memberlist KV store for hash rings"
+menuTitle: "Migrating from Consul to Memberlist"
+description: "Learn how to migrate from using Consul as KV store for hash rings to using memberlist without any downtime."
+weight: 30
+---
+
+# Migrating from Consul to Memberlist as KV store for hash rings without downtime
+
+Mimir Jsonnet uses `memberlist` as KV store for hash rings, unless disabled by using
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: false
+  }
+}
+```
+
+If you are still using Consul (ie. you have `memberlist_ring_enabled: false` in your Jsonnet, this was default before Grafana Mimir 2.2),
+and would like to migrate to `memberlist` without any downtime, you can follow instructions in this document.
+
+## Step 1: Enable memberlist and multi KV store.
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: true,
+    multikv_migration_enabled: true,
+  }
+}
+```
+
+Step 1 configures components to use `multi` KV store, with `consul` as primary and `memberlist` as secondary stores.
+This step requires rollout of all components.
+After applying this step all Mimir components will expose `/memberlist` page on HTTP admin interface, which can be used to check health of Memberlist cluster.
+
+## Step 2: Enable mirroring
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: true,
+    multikv_migration_enabled: true,
+    multikv_mirror_enabled: true,  // Changed in this step.
+  }
+}
+```
+
+In this step we enable writes to primary KV store (Consul) to be mirrored into secondary store (Memberlist).
+Applying this change will not cause restart of components.
+
+You can monitor following metrics to check if mirroring was enabled on all components and if it works correctly:
+
+- `cortex_multikv_mirror_enabled`
+- `rate(cortex_multikv_mirror_writes_total[1m])`
+- `rate(cortex_multikv_mirror_write_errors_total[1m])`
```

Comment:
I would just add few works about the expected result of these queries, to make it trivially easy to understand.